File: wenger-feature/src/main/java/io/github/rcarlosdasilva/wenger/feature/config/app/aliyun/OssProperties.java
package io.github.rcarlosdasilva.wenger.feature.config.app.aliyun;

import io.github.rcarlosdasilva.wenger.feature.config.AbleProperties;
import lombok.Data;

/**
 * @author <a href="mailto:rcarlosdasilva@qq.com">Dean Zhao</a>
 */
@Data
public class OssProperties extends AbleProperties {

  /**
   * OSS地址，默认外网地址
   */
  private String address = "https://oss-cn-shanghai.aliyuncs.com";
  /**
   * OSS Bucket
   */
  private String bucket;
  /**
   * 默认OSS文件缓存时间（Cache-Control: max-age），单位：秒，默认2,592,000秒（30天）
   */
  private long maxAge = 2592000L;

}

File: wenger-feature/src/main/java/io/github/rcarlosdasilva/wenger/feature/config/app/aliyun/OssProperties.kt
package io.github.rcarlosdasilva.wenger.feature.config.app.aliyun

import io.github.rcarlosdasilva.wenger.feature.config.AbleProperties

/**
 * @author [Dean Zhao](mailto:rcarlosdasilva@qq.com)
 */
class OssProperties : AbleProperties() {

  /**
   * OSS地址，默认外网地址
   */
  var address = "https://oss-cn-shanghai.aliyuncs.com"
  /**
   * OSS Bucket
   */
  var bucket: String? = null
  /**
   * 默认OSS文件缓存时间（Cache-Control: max-age），单位：秒，默认2,592,000秒（30天）
   */
  var maxAge = 2592000L

}
